src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/clsm_ensupd_upd_routines.F90

```diff
@@ -4403,6 +4406,194 @@ subroutine cat_enkf_increments(                               &
+       end if
+       
+       if (any(obs_param%varname == 'sfmc')) then
+          N_select_varnames = N_select_varnames + 1
+          select_varnames(N_select_varnames) = 'sfmc'     
+       end if
+
+       if (any(obs_param%varname == 'sfds')) then
+          N_select_varnames = N_select_varnames + 1
+          select_varnames(N_select_varnames) = 'sfds'     
+       end if
+       
+       ! Will get all species associated with Tb or sfds observations
+       
+       call get_select_species(                                           &
+            N_select_varnames, select_varnames(1:N_select_varnames),      &
+            N_obs_param, obs_param, N_select_species, select_species )         
+       
+       ! Determine which species are Tb
+       
+       call get_select_species(1, 'Tb', N_obs_param, obs_param, N_select_species_Tb, select_species_Tb )
+       
+       N_state_max = 7
+       
+       allocate( State_incr(N_state_max,N_ens)) 
+       allocate( State_lon( N_state_max      ))
+       allocate( State_lat( N_state_max      ))
+       
+       do kk=1,N_catd     
+          
+          N_state = 3    ! for now, assume only sfmc and/or sfds obs
+          
+          ! compute increments only snow-free and non-frozen tiles
+          
+          if ( (SWE_ensavg(kk) < SWE_threshold) .and. &
+               (tp1_ensavg(kk) > tp1_threshold)) then  
+             
+             ! find observations within halo around tile kk
+             
+             halo_minlon = tile_coord(kk)%com_lon - xcompact
+             halo_maxlon = tile_coord(kk)%com_lon + xcompact
+             halo_minlat = tile_coord(kk)%com_lat - ycompact
+             halo_maxlat = tile_coord(kk)%com_lat + ycompact
+             
+             ! simple approach to dateline issue (cut halo back to at most -180:180, -90:90)
+             ! - reichle, 28 May 2013
+             
+             halo_minlon = max(halo_minlon,-180.)
+             halo_maxlon = min(halo_maxlon, 180.)
+             halo_minlat = max(halo_minlat, -90.)
+             halo_maxlat = min(halo_maxlat,  90.)
+             
+             call get_ind_obs_lat_lon_box(                               &
+                  N_obs,            Observations,                        &
+                  halo_minlon, halo_maxlon, halo_minlat, halo_maxlat,    &
+                  N_select_species, select_species(1:N_select_species),  &
+                  N_selected_obs,   ind_obs )
+             
+             if (N_selected_obs>0) then
+                
+                ! Determine if Tb observations are present
+                
+                do ii = 1,N_select_species_Tb
+                   do jj = 1,N_selected_obs
+                      if (select_species_Tb(ii) == Observations(ind_obs(jj))%species) then
+                         found_Tb_obs = .true.
+                         exit
+                      end if
+                   end do
+                   if (found_Tb_obs) exit
+                end do
```

Comment:
We may want to simplify this block by teaching the `get_ind_obs_lat_lon_box()` (or the related `get_ind_obs()`) subroutine to (optionally) return a vector of the unique species numbers associated with the obs in ind_obs?  This would be useful when we're adding new multi-sensor update_types in the future.